package/kedro_viz/api/graphql.py

```diff
@@ -94,11 +94,12 @@ def get_all_runs() -> List[Run]:
     return runs
 
 
-def format_run_tracking_data(tracking_data: Dict) -> JSONObject:
+def format_run_tracking_data(tracking_data: Dict, show_diff: bool) -> JSONObject:
     """Convert tracking data in the front-end format.
 
     Args:
         tracking_data: JSON blob of tracking data for selected runs
+        show_diff : user specified boolean for comparison view
```

Comment:
I think this needs a bit more explanation about what it actually does, like you gave in the PR. Maybe:
```suggestion
        show_diff: whether to include metrics that are not defined for some of the selected runs in comparison view
```